homeassistant/components/lookin/light.py

```diff
@@ -0,0 +1,148 @@
+    lookin_data: LookinData = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+
+    for remote in lookin_data.devices:
+        if remote["Type"] != "03":
+            continue
+        uuid = remote["UUID"]
+
+        def _wrap_async_update(
+            uuid: str,
+        ) -> Callable[[], Coroutine[None, Any, Remote]]:
+            """Create a function to capture the uuid cell variable."""
+
+            async def _async_update() -> Remote:
+                return await lookin_data.lookin_protocol.get_remote(uuid)
+
+            return _async_update
+
+        coordinator = DataUpdateCoordinator(
+            hass,
+            LOGGER,
+            name=f"{config_entry.title} {uuid}",
+            update_method=_wrap_async_update(uuid),
+            update_interval=timedelta(
+                seconds=60
+            ),  # Updates are pushed (fallback is polling)
+        )
+        await coordinator.async_refresh()
+        device: Remote = coordinator.data
+
+        entities.append(
+            LookinLightEntity(
+                uuid=uuid,
+                device=device,
+                lookin_data=lookin_data,
+                coordinator=coordinator,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class LookinLightEntity(LookinPowerEntity, LightEntity):
+    """A lookin IR controlled light."""
+
+    _attr_supported_color_modes = {COLOR_MODE_ONOFF}
+    _attr_color_mode = COLOR_MODE_ONOFF
+
+    def __init__(
+        self,
+        uuid: str,
+        device: Remote,
+        lookin_data: LookinData,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Init the light."""
+        super().__init__(coordinator, uuid, device, lookin_data)
+        self._attr_is_on = False
+
+    @property
+    def _remote(self) -> Remote:
+        return cast(Remote, self.coordinator.data)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+        await self._async_send_command(self._power_on_command)
+        self._attr_is_on = True
+        self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the light."""
+        await self._async_send_command(self._power_off_command)
+        self._attr_is_on = False
+        self.async_write_ha_state()
+
+    def _update_from_status(self, status: str) -> None:
+        """Update media property from status.
+
+        1000
+        0 - 0/1 on/off
+        """
+        if len(status) != 4:
+            return
+        state = status[0]
+
+        self._attr_is_on = state == "1"
+
+    def _async_push_update(self, event: UDPEvent) -> None:
+        """Process an update pushed via UDP."""
+        LOGGER.debug("Processing push message for %s: %s", self.entity_id, event)
+        self._update_from_status(event.value)
+        self.coordinator.async_set_updated_data(self._remote)
+        self.async_write_ha_state()
```

Comment:
When we set the updated data on the coordinator the update callbacks of the coordinator will be called. So we don't need to update state explicitly here.

Reply:
Good call. I'm going to do another turn to cleanup this integration a bit in case the device only responds to some of the initial setup polls.  I'll adjust it there.

Reply:
https://github.com/home-assistant/core/pull/64106